dali/kernels/signal/fft/fft_postprocess_test.cu

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <random>
+#include "dali/kernels/common/copy.h"
+#include "dali/test/test_tensors.h"
+#include "dali/test/tensor_test_utils.h"
+#include "dali/kernels/scratch.h"
+
+#include "dali/kernels/signal/fft/fft_postprocess.cuh"
+
+namespace dali {
+
+
+template <>
+bool EqualEps::operator()<float2, float2>(const float2 &a, const float2 &b) const {
+  return std::abs(b.x - a.x) <= eps && std::abs(b.y - a.y) <= eps;
+}
+
+template <typename StorageBackend, int ndim, typename RandomGenerator>
+void UniformRandomFill(const TensorListView<StorageBackend, float2, ndim> &tlv,
+                       RandomGenerator &rng, float lo, float hi) {
+  static_assert(is_cpu_accessible<StorageBackend>::value,
+                "Function available only for CPU-accessible TensorListView backend");
+  auto dist = uniform_distribution(lo, hi);
+  auto gen = [&]() {
+    return float2{ dist(rng), dist(rng) };
+  };
+  Fill(tlv, gen);
+}
+
+namespace kernels {
+namespace signal {
+namespace fft_postprocess {
+
+TEST(FFTPostprocess, Norm2) {
+  norm2 f;
+  EXPECT_EQ(f(float2{0,0}), 0);
+  EXPECT_NEAR(f(float2{1,1}), sqrt(2), 1e-5f);
+  EXPECT_NEAR(f(float2{-1,1}), sqrt(2), 1e-5f);
+  EXPECT_NEAR(f(float2{-3,-4}), 5, 1e-5f);
+}
+
+TEST(FFTPostprocess, Norm2Square) {
+  norm2square f;
+  EXPECT_EQ(f(float2{0,0}), 0.0f);
+  EXPECT_EQ(f(float2{1,1}), 2.0f);
+  EXPECT_EQ(f(float2{-1,1}), 2.0f);
+  EXPECT_EQ(f(float2{-3,-4}), 25.0f);
+}
+
+TEST(FFTPostprocess, Power_dB) {
```

Comment:
Can you add a test case for non default cutoff value?

Reply:
Will do.

Reply:
Done.